.github/NativeBuildReport.java

```diff
@@ -0,0 +1,97 @@
+
+	@Option(names = "issueNumber", description = "The issue to update")
+	private Integer issueNumber;
+
+	@Option(names = "thisRepo", description = "The repository for which we are reporting the CI status")
+	private String thisRepo;
+
+	@Option(names = "runId", description = "The ID of the Github Action run for  which we are reporting the CI status")
+	private String runId;
+
+	@Override
+	public void run() {
+		try {
+			final boolean succeed = "success".equalsIgnoreCase(status);
+			if ("cancelled".equalsIgnoreCase(status)) {
+				System.out.println("Job status is `cancelled` - exiting");
+				System.exit(0);
+			}
+
+			System.out.println(String.format("The CI build had status %s.", status));
+
+			final GitHub github = new GitHubBuilder().withOAuthToken(token).build();
+			final GHRepository repository = github.getRepository(issueRepo);
+
+			final GHIssue issue = repository.getIssue(issueNumber);
+			if (issue == null) {
+				System.out.println(String.format("Unable to find the issue %s in project %s", issueNumber, issueRepo));
+				System.exit(-1);
+			} else {
+				System.out.println(String.format("Report issue found: %s - %s", issue.getTitle(), issue.getHtmlUrl().toString()));
+				System.out.println(String.format("The issue is currently %s", issue.getState().toString()));
+			}
+
+			if (succeed) {
+				if (issue != null  && isOpen(issue)) {
+					// close issue with a comment
+					final GHIssueComment comment = issue.comment(String.format("Build fixed:\n* Link to latest CI run: https://github.com/%s/actions/runs/%s", thisRepo, runId));
+					issue.close();
+					System.out.println(String.format("Comment added on issue %s - %s, the issue has also been closed", issue.getHtmlUrl().toString(), comment.getHtmlUrl().toString()));
+				} else {
+					System.out.println("Nothing to do - the build passed and the issue is already closed");
+				}
+			} else  {
+				if (isOpen(issue)) {
+					final GHIssueComment comment = issue.comment(String.format("The build is still failing:\n* Link to latest CI run: https://github.com/%s/actions/runs/%s", thisRepo, runId));
+					System.out.println(String.format("Comment added on issue %s - %s", issue.getHtmlUrl().toString(), comment.getHtmlUrl().toString()));
+				} else {
+					issue.reopen();
+					final GHIssueComment comment = issue.comment(String.format("Unfortunately, the build failed:\n* Link to latest CI run: https://github.com/%s/actions/runs/%s", thisRepo, runId));
+					System.out.println(String.format("Comment added on issue %s - %s, the issue has been re-opened", issue.getHtmlUrl().toString(), comment.getHtmlUrl().toString()));
+				}
+			}
+		}
+		catch (IOException e) {
+			throw new UncheckedIOException(e);
+		}
+	}
+
+	private static boolean isOpen(GHIssue issue) {
+		return (issue.getState() == GHIssueState.OPEN);
+	}
+
+	// this example implements Callable, so parsing, error handling and handling user
+	// requests for usage help or version help can be done with one line of code.
```

Comment:
Minor nitpick: this comment looks useless?

Reply:
Oooops, fixed :)